src/pybind/mgr/cephadm/ssh.py

```diff
@@ -61,7 +62,7 @@ def get_result(self, coro: Awaitable[T], timeout: Optional[int] = None) -> T:
         future = asyncio.run_coroutine_threadsafe(coro, self._loop)
         try:
             return future.result(timeout)
-        except asyncio.TimeoutError:
+        except (asyncio.TimeoutError, concurrent.futures._base.TimeoutError):
             # try to cancel the task before raising the exception further up
```

Comment:
Not sure if we can handle this without using concurrent.futures! ... async.io already provides the `asyncio.wait_for(future, timeout)` which supports a timeout. In case that's not possible and we have to go with the concurrent.futures I think we should rather capture `concurrent.futures.TimeoutError` as `_base` seems to be internal and not though to be used from outside.

Reply:
I like staying with the `run_coroutine_threadsafe` since it fits our use case of waiting for the future from outside of the event loop itself, but I've followed your second suggestion and removed the `_base` part of the exception type name after testing that it still worked without it.